frontend: network: Add support for Admin Network Policy resources

frozenprocess opened the pull request:
This change adds CRUD support for ANP.

## Summary

This PR adds/fixes [feature/bug] by [brief description of what the change does].

frontend: Add support for ANP resources which can be found under the Network section.


## Related Issue

Fixes #3575 

## Changes

- Added an ANP component to the UI
- Added a Kube Object to headlamp lib to represent ANP

[ x ] Having a dedicated sub menu link in the Network section for ANPs
[ x ] Create ANPs using the + button in the ANP link
[ x ] ANP List view
[ x ] ANP Detail view

## Steps to Test

1. Navigate to Network section
2. Click on Admin Network policies (alpha)
2a. create an ANP
2b. view your ANP
2c. delete your ANP
3. submit a feedback here :) 

## Screenshots (if applicable)


## Notes for the Reviewer

- [e.g., This touches the i18n layer, so please check language consistency.]

mazdakn commented:
@frozenprocess I personally think option 2 is better than 1. But is it possible to use icons before each subject selector for differentiation?

frozenprocess commented:
> @frozenprocess I personally think option 2 is better than 1. But is it possible to use icons before each subject selector for differentiation?

Yep, did some testing and it is possible to put an icon before each subject selector.

frozenprocess committed:
frontend: network: Add support for Admin Network Policy resources

ANP - Alpha support
- This change adds CRUD support for ANP.
- Added an ANP component to the UI
- Added a Kube Object to headlamp lib to represent ANP

frozenprocess committed:
Merge branch 'kubernetes-sigs:main' into anp

frozenprocess committed:
Merge branch 'kubernetes-sigs:main' into anp

frozenprocess committed:
Merge branch 'kubernetes-sigs:main' into anp

frozenprocess committed:
Merge branch 'kubernetes-sigs:main' into anp
